distribution/push_v1.go

```diff
@@ -241,7 +241,7 @@ func generateDependencyImages(l layer.Layer, dependenciesSeen map[layer.ChainID]
 		return nil, nil, nil
 	}
 
-	imageListForThisTag, parent, err = generateDependencyImages(l.Parent(), dependenciesSeen)
```

Comment:
`newV1DependencyImage` and `generateDependencyImages` should be changed not to return errors. Right now they always return a nil error.